presto-main/src/main/java/com/facebook/presto/sql/planner/GroupingOperationRewriter.java

```diff
@@ -125,12 +125,15 @@ static long calculateGrouping(long groupId, List<Integer> columns, List<List<Int
         long grouping = (1L << columns.size()) - 1;
 
         List<Integer> groupingSet = groupingSetDescriptors.get(toIntExact(groupId));
-        for (Integer groupingColumn : groupingSet) {
-            int index = columns.indexOf(groupingColumn);
-            if (index != -1) {
+
+        for (int index = 0; index < columns.size(); index++) {
+            int column = columns.get(index);
+
+            if (groupingSet.contains(column)) {
```

Comment:
`groupingSet` is not a set, so a problem for (absurdly?) long `grouping(..)` list

Reply:
Aha! This commit was meant to appear last, after groupingSet is made a set in "compute grouping sets during planning"